### PR TITLE
Rename API group name to "quarks.cloudfoundry.org"

### PR DESCRIPTION
**Pull request by manno**
Requires: https://github.com/cloudfoundry-incubator/quarks-job/pull/19
Requires: https://github.com/cloudfoundry-incubator/cf-operator-ci/pull/99

[#169095318](https://www.pivotaltracker.com/story/show/169095318)

### CODE REVIEW

File: pkg/kube/client/clientset/versioned/fake/clientset_generated.go [hunk 1]
```diff
@@ -34,7 +34,7 @@ func NewSimpleClientset(objects ...runtime.Object) *Clientset {
 		}
 	}
 
-	cs := &Clientset{tracker: o}
+	cs := &Clientset{}
```
Comment (manno) [inline pkg/kube/client/clientset/versioned/fake/clientset_generated.go]: This is not intentional, I regenerated those files and that was the only change. Probably doesn't matter?

File: pkg/kube/controllers/boshdeployment/validating_webhook.go [hunk 2]
```diff
@@ -34,7 +34,7 @@ func NewBOSHDeploymentValidator(log *zap.SugaredLogger, config *config.Config) *
 		Rules: []admissionregistrationv1beta1.RuleWithOperations{
 			{
 				Rule: admissionregistrationv1beta1.Rule{
-					APIGroups:   []string{"fissile.cloudfoundry.org"},
+					APIGroups:   []string{"quarks.cloudfoundry.org"},
```
Comment (qu1queee) [inline pkg/kube/controllers/boshdeployment/validating_webhook.go]: any reason why you are not using the https://github.com/cloudfoundry-incubator/quarks-utils/blob/af677c684e378cce33c908f8ef640ed39b60e78f/pkg/names/apigroup.go#L4 in here?

File: pkg/kube/controllers/boshdeployment/validating_webhook.go [hunk 3]
```diff
@@ -46,7 +46,7 @@ func NewBOSHDeploymentValidator(log *zap.SugaredLogger, config *config.Config) *
 			},
 		},
 		Path: "/validate-boshdeployment",
-		Name: "validate-boshdeployment.fissile.cloudfoundry.org",
+		Name: "validate-boshdeployment.quarks.cloudfoundry.org",
```
Comment (qu1queee) [inline pkg/kube/controllers/boshdeployment/validating_webhook.go]: same as above

File: pkg/kube/controllers/extendedstatefulset/pod_webhook.go [hunk 4]
```diff
@@ -34,7 +34,7 @@ func NewExtendedStatefulsetPodMutator(log *zap.SugaredLogger, config *config.Con
 			},
 		},
 		Path: "/mutate-pods",
-		Name: "mutate-pods.fissile.cloudfoundry.org",
+		Name: "mutate-pods.quarks.cloudfoundry.org",
```
Comment (qu1queee) [inline pkg/kube/controllers/extendedstatefulset/pod_webhook.go]: any reason why you are not using the https://github.com/cloudfoundry-incubator/quarks-utils/blob/af677c684e378cce33c908f8ef640ed39b60e78f/pkg/names/apigroup.go#L4 in here?

File: testing/catalog.go [hunk 5]
```diff
@@ -637,7 +637,7 @@ func (c *Catalog) NodePortService(name, ig string, targetPort int32) corev1.Serv
 		Spec: corev1.ServiceSpec{
 			Type: corev1.ServiceTypeNodePort,
 			Selector: map[string]string{
-				"fissile.cloudfoundry.org/instance-group-name": ig,
+				"quarks.cloudfoundry.org/instance-group-name": ig,
```
Comment (qu1queee) [inline testing/catalog.go]: same as previous comments.

**Pull request merged**